### ISSUE TITLE
EventDateTime decoder will not decode when date is null

### ISSUE THREAD

**Issue by amit115532** [GoogleCloudPlatform/elixir-google-api#75]
When creating an event in the calendar api the response contains an Event which contains EventDateTime.
Inside it, thers an optional field called "date"
for some reason, EventDateTime decoder will try to deserialize the date explicitly 
```
defimpl Poison.Decoder, for: GoogleApi.Calendar.V3.Model.EventDateTime do
  import GoogleApi.Calendar.V3.Deserializer
  def decode(value, options) do
    value
    |> deserialize(:"date", :date, nil, options)
  end
end
```
but there is no deserialize function with nil 3rd argument in calendar/deserializer.ex

By adding one, I managed to get the code to work.
Thats not a fix though, I need the generator to do that automatically.. And im not sure how the code works.

**Comment by chingor13** [comment]
Thanks for reporting this. We fixed this issue previously in the mustache template itself but hadn't rebuilt the calendar client to address it in that client. We'll rebuild and republish.

**Comment by chingor13** [comment]
The calendar client should now be available on hex as 0.0.2.
